ldap-authenticator/src/main/java/org/xwiki/contrib/ldap/LDAPProfileXClass.java

```diff
@@ -235,32 +235,52 @@ public boolean updateLDAPObject(XWikiDocument userDocument, String dn, String ui
     /**
      * Search the XWiki storage for a existing user profile with provided LDAP user uid stored.
      * <p>
-     * If more than one profile is found the first one in returned and an error is logged.
-     * 
+     * If more than one profile is found the first one is returned and an error is logged.
+     *
      * @param uid the LDAP unique id.
      * @return the user profile containing LDAP uid.
      */
     public XWikiDocument searchDocumentByUid(String uid)
+    {
+        return searchDocumentByLdapProfileClassAttribute(LDAP_XFIELD_UID, uid);
+    }
+
+    /**
+     * Search XWiki storage for a existing user profile with provided LDAP dn stored.
+     * <p>
+     * If more than one profile is found the first one is returned and an error is logged.
+     *
+     * @param dn the LDAP DN.
+     * @return the user profile containing LDAP dn.
+     * @since 9.4.6
```

Comment:
I think I would make it 9.5, feels more like an improvement/new feature than bugfixing.

Reply:
ok, done